Course/Course/Domain/CourseInteractor.swift

```diff
@@ -64,6 +64,7 @@ public class CourseInteractor: CourseInteractorProtocol {
     }
     
     public func blockCompletionRequest(courseID: String, blockID: String) async throws {
+        NotificationCenter.default.post(name: .onblockCompletionRequested, object: courseID)
```

Comment:
Wouldn't this be an overhead by refreshing enrollments on each component completed? 
What do you think about about the approach, mark it that enrollments needs to be refreshed and refresh the enrollments when user move to either course home or dashboard screen?